aiida_mlip/parsers.py

```diff
@@ -1,58 +1,136 @@
 """
 Parsers provided by aiida_mlip.
-
-Register parsers via the "aiida.parsers" entry point in setup.json.
 """
 
+from pathlib import Path
+
+from ase.io import read
+import numpy as np
+
 from aiida.common import exceptions
 from aiida.engine import ExitCode
-from aiida.orm import SinglefileData
+from aiida.orm import Dict, SinglefileData
 from aiida.parsers.parser import Parser
 from aiida.plugins import CalculationFactory
 
-DiffCalculation = CalculationFactory("mlip")
+Singlepointcalc = CalculationFactory("janus.sp")
 
 
-class DiffParser(Parser):
+def convert_numpy(dictionary: dict) -> dict:
+    """
+    A function to convert numpy ndarrays in dictionary into lists.
+
+    Parameters
+    ----------
+    dictionary : dict
+        A dictionary with numpy array values to be converted into lists.
+
+    Returns
+    -------
+    dict
+        Converted dictionary.
+    """
+    for key, value in dictionary.items():
+        if isinstance(value, np.ndarray):
+            dictionary[key] = value.tolist()
+    return dictionary
+
+
+class SPParser(Parser):
     """
     Parser class for parsing output of calculation.
+
+    Parameters
+    ----------
+    node : aiida.orm.nodes.process.process.ProcessNode
+        ProcessNode of calculation.
+
+    Methods
+    -------
+    __init__(node: aiida.orm.nodes.process.process.ProcessNode)
+        Initialize the SPParser instance.
+
+    parse(**kwargs: Any) -> int:
+        Parse outputs, store results in the database.
+
+    Returns
+    -------
+    int
+        An exit code.
+
+    Raises
+    ------
+    exceptions.ParsingError
+        If the ProcessNode being passed was not produced by a Singlepointcalc.
     """
 
     def __init__(self, node):
         """
-        Initialize Parser instance
-
-        Checks that the ProcessNode being passed was produced by a DiffCalculation.
+        Check that the ProcessNode being passed was produced by a Singlepointcalc.
 
-        :param node: ProcessNode of calculation
-        :param type node: :class:`aiida.orm.nodes.process.process.ProcessNode`
+        Parameters
+        ----------
+        node : aiida.orm.nodes.process.process.ProcessNode
+            ProcessNode of calculation.
         """
         super().__init__(node)
-        if not issubclass(node.process_class, DiffCalculation):
-            raise exceptions.ParsingError("Can only parse DiffCalculation")
+        if not issubclass(node.process_class, Singlepointcalc):
+            raise exceptions.ParsingError("Can only parse Singlepointcalc")
 
+    # pylint: disable=too-many-locals
```

Comment:
Not sure we want this, much as I hate to say it, I kind of agree with pylint here. You have a lot of single use variables which don't really add much clarity to me.